### PR TITLE
Refactor Shop Entities

### PR DESCRIPTION
**Pull request by lucca65**
## What issue does this PR close
None, but its part of a major effort to refactor our backend entities, for #113. 

## Changes Proposed ( a list of new changes introduced by this PR)
Update our GraphQL Shop schemas do follow changes introduced by the past few months

- Rename tables 
- Add a new context for Shop
- Add a new type module only for shop operations
- Remove old/unused shop queries
- Start using reusable queries
- Added two new community counters

## How to test ( a list of instructions on how to test this PR)
- Run `mix test`.
- Use all the frontend features necessary, such as creating new Products, viewing your notification history for orders, check the listing of shop

### CODE REVIEW

File: lib/cambiatus/shop/order.ex [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
     field(:units, :integer)
 
+    field(:created_block, :integer)
+    field(:created_tx, :string)
+    field(:created_eos_account, :string)
+    field(:created_at, :utc_datetime)
+
     belongs_to(:community, Community, references: :symbol, type: :string)
-    belongs_to(:sale, Sale)
+    belongs_to(:product, Product)
```
Comment (amiskov) [inline lib/cambiatus/shop/order.ex:25]: In the `orders` table the field for the product ID called `sale_id`. So, we have to rename the column to `product_id` to use `belongs_to` this way or use `foreign_key: :sale_id` parameter. I think renaming is the way to go.

**Review by amiskov — CHANGES_REQUESTED** [review]
The code looks much cleaner now! I'm very happy that you are doing this refactoring :)

Found a couple of things to fix in the `Order` module. I found them by visiting the Notification page (the bell at the top right) from the related [frontend branch](https://github.com/cambiatus/backend/pull/119). The backend couldn't find `product_id`.

File: lib/cambiatus/shop/order.ex (hunk 1 above)
Comment (amiskov) [reply lib/cambiatus/shop/order.ex:25]: It will raise an error while trying to request orders e.g. from the Notification page.

Comment (lucca65) [reply lib/cambiatus/shop/order.ex:25]: Nice catch @amiskov! thanks for noticing that! I've renamed the column

**Comment by lucca65** [comment]
@amiskov I've added a few more queries for some fixes I'm doing based on an user report, I thought that the work of having several PRs for those things would be too much trouble.

**Comment by amiskov** [comment]
@lucca65 It looks like we also need to update the event-source since it uses the `db.sales` table for inserting/updating products. E.g. in [this file](https://github.com/cambiatus/event-source/blob/master/src/updaters/community.js).

You can check the bug by creating/updating the product in the Shop. There will be nothing changed on the frontend and if you check the `pm2 logs event-source` on Staging you'll see an error pointing to the [updaters/community.js](https://github.com/cambiatus/event-source/blob/master/src/updaters/community.js#L193) from the `event-source` repo.

**Comment by lucca65** [comment]
sure, it will be done in the event-source repo!

**Comment by lucca65** [comment]
done and fixed on staging: https://github.com/cambiatus/event-source/pull/29

**Review by amiskov — APPROVED** [review]
Tudo bem para mim! 👍

**Pull request merged**